### DISCUSSION TITLE
协议驱动：采用 数据是数据 、UI 是 UI，在后端更方便。

### DISCUSSION THREAD

**Discussion by kaoding** [alibaba/formily#2440] [Q&A 2.x] [answered]
**react-jsonschema-form的解法是，数据是数据，UI 是 UI，这样的好处是，各个协议都是非常纯净的协议**，
这个在后端很有用，如果将数据和UI合在一起，会增加后端心智成本。

1.  后端就放在一起，使用数据是从合在一起的协议中提取数据使用。
2.  如果后端是分开的，在给前端时候合好给前端。

综合考虑了前端的情况，欢迎大家讨论

**Comment by janryWang** [comment, marked as answer]
拆分存在3个问题：
1. 对搭建非常不友好
2. 对研发效率非常不友好，一个协议描述UI+逻辑，一个协议描述结构+类型，两边切来切去，简单场景还好，复杂场景，地狱般的难受
2. Formily Schema定位是描述视图模型的Schema，不是描述后端领域模型的Schema，两者本身就不是幂等的，所以后端领域模型用标准json-schema没任何问题，但不推荐用formily schema，可以是相互导出转换关系，比如从后端领域模型schema导出成formily schema，也可以将formily schema导出成后端领域模型schema，但就不应该是共享同一份的关系
